app/controllers/chat_controller.rb

```diff
@@ -39,7 +39,7 @@ def enable_chat
     elsif chat_channel
       return render_json_error I18n.t("chat.already_enabled")
     else
-      chat_channel = ChatChannel.new(chatable: @chatable)
+      chat_channel = ChatChannel.for(@chatable).new(chatable: @chatable)
```

Comment:
Yes I’m not completely satisfied either with this API, I had another version which was instantiating a new object but then it was a bit weird with the `create!` call we have elsewhere. Well I’ll take another look and see if it can be improved :grin: